### ISSUE TITLE
Truffle Compile re-downloads solc even after running Truffle Obtain

### ISSUE THREAD

**Issue by adrianmcli** [trufflesuite/truffle#1835]
## Issue

Truffle Compile re-downloads solc even after running Truffle Obtain. Do note that we use `truffle` to run `truffle obtain`.

But in our tests we use `truffle-compile` by itself (as in we did `npm install truffle-compile` and used it with `require("truffle-compile")`).

## Steps to Reproduce

This is the .travis.yml used [here](https://github.com/trufflesuite/drizzle-utils/blob/master/.travis.yml):

```yml
language: node_js
node_js:
  - "node"
  - "--lts"
cache:
  yarn: true
  directories:
    - node_modules
before_script:
  - "truffle obtain --solc=0.5.2"
```

And this is the `truffle-compile` usage: https://github.com/trufflesuite/drizzle-utils/blob/master/packages/test-chain/compile.js#L3

## Expected Behavior

Since `truffle obtain` is already run, it shouldn't try to download the solidity compiler again.

## Actual Results

Check out this build:

https://travis-ci.org/trufflesuite/drizzle-utils/jobs/508220795

Note that on line 468, it already downloaded the compiler. But on line 539 it tries to do it again. It's able to fetch it in time for this build, but not in other builds.

**Comment by CruzMolina** [comment]
Looking into this, I wonder if your `truffle-compile` dependency needs to be updated...

**Comment by CruzMolina** [comment]
So, I updated `truffle-compile` to latest (4.0.8) on the `update-truffle-compile` branch in drizzle-utils, and I'm not sure whether it fixed the problem (the re-downloading output went away) or caused another one as the build seems to completely stall and then timeout.

https://travis-ci.org/trufflesuite/drizzle-utils/builds/508644454

**Comment by CruzMolina** [comment]
Oh... I _think_ `truffle-compile` comes loaded w/ solc 0.5.0.

An earlier version of `truffle-compile` was wonky in that it either didn't cache solc properly or it continued to try and download a specified version of solc anyway (even if it was already cached).

I wonder if the fix for that compartmentalized `truffle-compile` so that by requiring and using it in the tests, there's some error somewhere being thrown, but unhandled from `truffle-compile` trying to use solc 0.5.0. Could be why the build as well as testing locally seems to hang. 🤔

**Comment by adrianmcli** [comment]
@CruzMolina Thanks for looking into this. What do you suggest we do?

**Comment by CruzMolina** [comment]
@adrianmcli short-term solution would be to use `truffle` itself to compile, in the meanwhile I'll be looking into de-coupling `truffle-compile` again.

**Comment by CruzMolina** [comment]
Okay, so where I'm at in this issue appears to be the same problem as #1813 .

Currently working on a fix that both gets around the known `sudo` installation issue and` babel` binding itself to node's `require`.

**Comment by CruzMolina** [comment]
To recap, this issue directly relates to an older version of `truffle-compile` (maybe `4.0.0`) that improperly cached `solc` and would attempt to re-download after using `truffle obtain`. This is now fixed on the latest version of `truffle-compile`, but there is another issue outlined in #1813 .

Keeping this open for now as a reminder, hoping to have the solution for #1813 merged and released in `truffle-compile: 4.0.9` soon.

**Comment by gnidan** [comment]
This must be released, then? Closing until I hear otherwise!

**Comment by adrianmcli** [comment]
@CruzMolina Just wanted to ping you to check if this has indeed been fixed?

**Comment by CruzMolina** [comment]
hey @adrianmcli , this should be fixed in `truffle-compile`: `4.0.10`.

 🍾

**Comment by adrianmcli** [comment]
Thanks, I'll give it a shot!
